Sources/Polygon.swift

```diff
@@ -278,7 +281,74 @@ internal extension Collection where Element == Polygon {
     func tessellate() -> [Polygon] {
         flatMap { $0.tessellate() }
     }
-
+	
+	func detessellate() -> [Polygon] {
+		// TODO:
+		// - Make map of connected Polygons
+		
+		var edgePolyMap : [[Double]:[Polygon]] = [:]
+		
+		// Map the polys by edge
+		forEach { polygon in
+			// Likely 3 edges
+			undirectedEdgeKeys(polygon).forEach {
+				if edgePolyMap[$0] == nil { edgePolyMap[$0] = [] }
+				edgePolyMap[$0]?.append(polygon)
+			}
+		}
+		
+		// Join coplanar polygons
+		edgePolyMap.keys.forEach {
+			let joinEdge = $0
+			let polygons = edgePolyMap[joinEdge]
+			if polygons?.count == 2 {
+				if polygons![0].plane.isEqual(to: polygons![1].plane) {
+					let prejoinedPoly1 = polygons![0]
+					let prejoinedPoly2 = polygons![1]
+					let joinedPoly = prejoinedPoly1.join(unchecked: prejoinedPoly2, ensureConvex: false)
+					if let joinedPoly = joinedPoly {
+						edgePolyMap.removeValue(forKey: joinEdge)
+						undirectedEdgeKeys(joinedPoly).forEach {
+							edgePolyMap[$0]?.removeAll(where: { p in p == prejoinedPoly1 || p == prejoinedPoly2 })
+							if edgePolyMap[$0] == nil { edgePolyMap[$0] = [] }
+							edgePolyMap[$0]?.append(joinedPoly)
+						}
+					}
+				}
+			}
+		}
+		
+		return edgePolyMap.values.flatMap { $0 }
+	}
+	
+	private func undirectedEdgeKeys(_ p: Polygon) -> [[Double]] {
+		var edges: [[Double]] = []
+		for i in 0..<p.vertices.count {
+			let v1 = p.vertices[i]
+			let v2 = p.vertices[i-1 < 0 ? p.vertices.count - 1 : i-1]
+			let undirectedEdge = v1.position.length < v2.position.length ? [v1,v2] : [v2, v1]
+			edges.append([
+				undirectedEdge[0].position.x, undirectedEdge[0].position.y, undirectedEdge[0].position.z,
```

Comment:
Why create an intermediate array with 6 double values instead of just directly storing the two vectors here?

Reply:
I assumed that Vectors equate based on reference, not value. Didn't take the time to test this.

Reply:
No, all the types in Euclid are value types.